packages/web/src/components/result/ReactiveList.js

```diff
@@ -152,15 +164,24 @@ class ReactiveList extends Component {
 		}
 
 		// called when page is changed
-		if (this.props.pagination && this.state.isLoading) {
-			if (nextProps.onPageChange) {
-				nextProps.onPageChange();
-			} else {
-				window.scrollTo(0, 0);
+		if (this.props.pagination) {
+			if (this.state.isLoading) {
+				if (nextProps.onPageChange) {
+					nextProps.onPageChange({
+						currentPage: this.state.currentPage + 1,
+						totalPages: this.state.totalPages,
+					});
+				} else {
+					window.scrollTo(0, 0);
+				}
+				this.setState({
+					isLoading: false,
+				});
+			} else if (this.props.currentPage !== nextProps.currentPage
```

Comment:
Why do we have this in an `else if` rather than having it as an independent `if` block? 🤔